src/main/java/ca/on/oicr/pineryreports/reports/impl/LibrariesBillingReport.java

```diff
@@ -299,7 +307,7 @@ protected int getRowCount() {
   }
 
   private String[] makeDetailedRow(DetailedObject obj) {
-    String[] row = new String[getColumns().size()];
+    String[] row = new String[getDetailedHeadings().size()];
```

Comment:
Instead of changing this, should add another blank column definition to `getColumns` to keep it consistent. Not sure if it could cause any problems, but `getColumns` is intended to return a column definition for each column in the spreadsheet

Reply:
I found this confusing because getColumns doesn't return the columns in the part of the spreadsheet I'm editing. 
getColumns returns "Study Title", "Library Kit", "Count (%s - %s)", and then two blank columns (?)
but the columns Paul described in the ticket were in 'getDetailedHeadings' and when using getColumns on line 310 the sizes didn't match, I thought this made more sense, and unless those blank columns store still have data in the first part of the spreadsheet, we could drop the blank columns

Reply:
`TableReport` was designed to create a spreadsheet with a single table, but it's being extended to create spreadsheets containing multiple tables. A better design would be to create a new abstract report class for creating a spreadsheet with multiple tables. Instead, to make `TableReport` work for this, `getColumns` should return a column definition for each column in the widest of those tables, but only have a heading for the ones in the first table. Headings in other tables must be added as rows later.

Did you try generating a report with your changes? I would think at best, the csv would be poorly formed, as not all rows would have the same number of columns.

Reply:
I generated a report with my changes and it displays just fine. I can send it to you on Slack if you'd like

Reply:
Open in a text editor and check if every row has the same number of columns. If not, it's bad csv, even if Excel displays it fine.

Reply:
Best I could find on non-rectangular csv is the RFC which says rows "should" (not "must") be all the same width: https://tools.ietf.org/html/rfc4180#page-2

Still, I refactored getColumns() so that all the rows are the same length and the list of summary table headings isn't pulling double-duty as a width definition with those unexplained blanks.

Reply:
Many tools that read csv fail when rows have different lengths, and "should" indicates best practice anyway. The Javadoc for `TableReport` does also state that `getRow` "Must return an array of the same size and in the same order as getColumns"